orderbook/src/fee.rs

```diff
@@ -140,6 +158,29 @@ impl MinFeeCalculator {
     }
 }
 
+async fn get_sell_amount(
+    sell_token: H160,
+    buy_token: H160,
+    amount: U256,
+    kind: OrderKind,
+    price_estimator: &dyn PriceEstimating,
+) -> Result<U256> {
+    Ok(match kind {
+        OrderKind::Buy => U256::from_f64_lossy(
+            amount.to_f64_lossy()
+                * price_estimator
+                    .estimate_price(sell_token, buy_token)
+                    .await?,
+        ),
+        OrderKind::Sell => amount,
+    })
+}
+
+fn round_amount(amount: U256) -> U256 {
+    let order_of_magnitude = amount.to_string().len() - 1;
+    U256::exp10(order_of_magnitude) * 5
+}
```

Comment:
> Amounts are converted into sell amounts and rounded to the midpoint of their order of magnitude (e.g. 110 -> 500, etc.) to allow more cache hits (I'm not sure we actually want this).

I am not sure either. The way this is implemented here means that all possible sell amounts come in factors of 10 like 5, 50, 500, 5000 etc. Since it rounds a sell amount can be changed by up to a factor of 5. This is nice for the cache but has the drawback that fee is less accurate.
We could make it more accurate while keeping this type of rounding by using exponents of 2 instead of 10 with U256's `bits` and `pow` methods.

Reply:
After thinking about this a bit more, it might be better for now to not optimize cache hits and only add "bucketing" if we see we are getting too many requests. I think it's generally safer and easier to debug to not do the buy_amount -> sell_amount order conversion (as prices might change between the fee request and order placement) and store a model in the DB that is closer to the actual request. I will rework #364 to store amount and order kind instead of always assuming sell_orders.

I do like the suggestion of using power of two instead of power of tens.